subprojects/docs/src/snippets/tasks/incrementalBuild-incrementalBuildAdvanced/kotlin/build.gradle.kts

```diff
@@ -20,10 +20,10 @@ dependencies {
 
 // tag::custom-task-class[]
 tasks.register<ProcessTemplates>("processTemplates") {
-    templateEngine.set(TemplateEngineType.FREEMARKER)
-    templateData.name.set("test")
-    templateData.variables.set(mapOf("year" to "2012"))
-    outputDir.set(file(layout.buildDirectory.dir("genOutput")))
+    templateEngine = TemplateEngineType.FREEMARKER
+    templateData.name = "test"
+    templateData.variables = mapOf("year" to "2012")
+    outputDir = file(layout.buildDirectory.dir("genOutput"))
```

Comment:
❓ Why do we pass this through `file()` here instead of calling `.get().getAsFile()` or something?

Reply:
Good point, I don't know why we wrap that to a file().
Actually it works with just:
```
outputDir = layout.buildDirectory.dir("genOutput")
```

Reply:
Updated the snippet

Reply:
Great. This pattern appeared in many places, wonder if we can fix them all.

Reply:
Opened issue: https://github.com/gradle/gradle/issues/25789